src/translation.rs

```diff
@@ -1,47 +1,38 @@
-use crate::{Capability, Error, Namespace, RESOURCE_PREFIX};
-
-use std::collections::BTreeMap;
-use std::fmt::Write;
+use crate::{Capability, Error, RESOURCE_PREFIX};
 
 use iri_string::types::UriString;
+use serde::{Deserialize, Serialize};
 use siwe::Message;
 
 /// Extract the encoded capabilities from a SIWE message.
-pub fn extract_capabilities(message: &Message) -> Result<BTreeMap<Namespace, Capability>, Error> {
+pub fn extract_capabilities(message: &Message) -> Result<Option<Capability>, Error> {
     message
         .resources
         .iter()
-        .filter(|res| res.as_str().starts_with(RESOURCE_PREFIX))
-        .map(<(Namespace, Capability)>::from_resource)
-        .collect()
+        .last()
+        .map(Capability::from_resource)
+        .transpose()
 }
 
 /// Generate a ReCap statement from capabilities and URI (delegee).
-pub fn capabilities_to_statement(
-    capabilities: &BTreeMap<Namespace, Capability>,
+pub fn capabilities_to_statement<NB>(
```

Comment:
Should probably be a method now that it's a local type. Otherwise, it should be renamed `capability_to_statement`?

Reply:
done 👍